File: modulo05_for/divisores_com_for.js
import { get_positive_number, print } from '../utils/io_utils.js'

function main(){

  const numero = get_positive_number('Digite um número: ')
  const valores = obter_dividores(numero)
  print(`O número ${numero} tem ${valores.length} divisores`)

  const result = `São eles: ${valores.join(', ')}`
  print(result)

}

function obter_dividores(numero){
  const divisores = []

  for(let cand = 1; cand <= numero; cand = cand + 1){
    if (numero % cand === 0){
      divisores.push(cand)
    }
  }

  return divisores
}

main()

File: modulo05_for/divisores_com_for.py
def main():
    # Entrada
    numero = int(input('> Digite um número:\n> '))

    # Processamento
    valores = obter_divisores(numero)

    # Saídas
    result = f'O número {numero} tem {len(valores)} divisdores'
    print(result)
    # print('São eles: ', ', '.join(list(map(str, valores))))
    for item in valores:
        print(item)


def obter_divisores(numero: int):
    divisores = []

    for i in range(1, numero + 1):
        if numero % i == 0:
            divisores.append(i)

    return divisores


# run
main()
